src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -403,9 +403,33 @@ public String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> r
             return "";
         }
 
-        boolean isContainsNonEmptyResponse = false; // we will only show stats if there is at least one nonempty response
-
         Map<String, Integer> answerFrequency = new LinkedHashMap<String, Integer>();
+        int numChoicesSelected = getQuestionResultStatistics(responses, answerFrequency);
+        if (numChoicesSelected == -1) {
+            return "";
+        }
+        DecimalFormat df = new DecimalFormat("#.##");
+
+        StringBuilder fragments = new StringBuilder();
+        for (Entry<String, Integer> entry : answerFrequency.entrySet()) {
+            fragments.append(Templates.populateTemplate(FormTemplates.MCQ_RESULT_STATS_OPTIONFRAGMENT,
+                                Slots.MCQ_CHOICE_VALUE, entry.getKey(),
+                                Slots.COUNT, entry.getValue().toString(),
+                                Slots.PERCENTAGE,
+                                df.format(100 * divideOrReturnZero((double) entry.getValue(), numChoicesSelected))));
+
+        }
+        //Use same template as MCQ for now, until they need to be different.
+        return Templates.populateTemplate(FormTemplates.MCQ_RESULT_STATS, Slots.FRAGMENTS, fragments.toString());
+    }
+
+    /**
+     * Check if there is a non empty response.
+     * @return -1 if there is no empty response else number of empty response.
+     */
+    private int getQuestionResultStatistics(
```

Comment:
The method name does not match its method signature. If it really is about getting statistics, I'd expect some kind of statistics class or object as the return type, not an integer or "-1" if there are no empty responses (what does empty response have to do with it?).